pkg/operation/operation.go

```diff
@@ -420,56 +419,57 @@ func (o *Operation) ReportShootProgress(ctx context.Context, stats *flow.Stats)
 	)
 
-	newShoot, err := kutil.TryUpdateShootStatus(ctx, o.K8sGardenClient.GardenCore(), retry.DefaultRetry, o.Shoot.Info.ObjectMeta,
-		func(shoot *gardencorev1beta1.Shoot) (*gardencorev1beta1.Shoot, error) {
-			if shoot.Status.LastOperation == nil {
-				return nil, fmt.Errorf("last operation of Shoot %s/%s is unset", shoot.Namespace, shoot.Name)
-			}
-			if shoot.Status.LastOperation.LastUpdateTime.After(lastUpdateTime.Time) {
-				return nil, fmt.Errorf("last operation of Shoot %s/%s was updated mid-air", shoot.Namespace, shoot.Name)
-			}
-			if description != "" {
-				shoot.Status.LastOperation.Description = description
-			}
-			shoot.Status.LastOperation.Progress = progress
-			shoot.Status.LastOperation.LastUpdateTime = lastUpdateTime
-			return shoot, nil
-		})
-	if err != nil {
+	if err := func() error {
+		patch := client.StrategicMergeFrom(shoot.DeepCopy())
+
+		if shoot.Status.LastOperation == nil {
+			return fmt.Errorf("last operation of Shoot %s/%s is unset", shoot.Namespace, shoot.Name)
+		}
+		if shoot.Status.LastOperation.LastUpdateTime.After(lastUpdateTime.Time) {
+			return fmt.Errorf("last operation of Shoot %s/%s was updated mid-air", shoot.Namespace, shoot.Name)
+		}
+		if description != "" {
+			shoot.Status.LastOperation.Description = description
+		}
+		shoot.Status.LastOperation.Progress = progress
+		shoot.Status.LastOperation.LastUpdateTime = lastUpdateTime
+
+		return o.K8sGardenClient.Client().Status().Patch(ctx, shoot, patch)
+	}(); err != nil {
 		o.Logger.Errorf("Could not report shoot progress: %v", err)
 		return
 	}
-
-	o.Shoot.Info = newShoot
 }
 
 // CleanShootTaskErrorAndUpdateStatusLabel removes the error with taskID from the Shoot's status.LastErrors array.
 // If the status.LastErrors array is empty then status.LastErrors is also removed. It also re-evaluates the shoot status
 // in case the last error list is empty now, and if necessary, updates the status label on the shoot.
 func (o *Operation) CleanShootTaskErrorAndUpdateStatusLabel(ctx context.Context, taskID string) {
-	updatedShoot, err := kutil.TryUpdateShootStatus(ctx, o.K8sGardenClient.GardenCore(), retry.DefaultRetry, o.Shoot.Info.ObjectMeta,
-		func(shoot *gardencorev1beta1.Shoot) (*gardencorev1beta1.Shoot, error) {
-			shoot.Status.LastErrors = gardencorev1beta1helper.DeleteLastErrorByTaskID(o.Shoot.Info.Status.LastErrors, taskID)
-			return shoot, nil
-		},
-	)
-	if err != nil {
-		o.Logger.Errorf("Could not update shoot's %s/%s last errors: %v", o.Shoot.Info.Namespace, o.Shoot.Info.Name, err)
+	shoot := o.Shoot.Info
+
+	statusPatch := client.MergeFrom(shoot.DeepCopy())
+	// LastErrors doesn't have patchMergeKey as TaskID is optional, so strategic merge wouldn't make a difference.
+	// This will effectively overwrite the whole LastErrors array. Though, this should be fine, as this controller is
+	// supposed to be the exclusive owner of this field. Even if some other goroutine writes to LastErrors concurrently,
```

Comment:
Out of curiosity: Which other goroutines do you have in mind here?

Reply:
Ok, I see this comment is confusing, as it is based on a wrong assumption on my side. 
Just removed it.